packages/arb-ts/token_lists/generate_json.ts

```diff
@@ -25,21 +25,72 @@ import { TokenInfo, TokenList, schema } from '@uniswap/token-lists'
 
 import { StandardArbERC20__factory } from '../src/lib/abi/factories/StandardArbERC20__factory'
 import { ERC20__factory } from '../src/lib/abi/factories/ERC20__factory'
+import { ERC20 } from '../src/lib/abi//ERC20'
 
 import { instantiateBridge } from '../scripts/instantiate_bridge'
 
 const ajv = new Ajv()
 addFormats(ajv)
 const validate = ajv.compile(schema)
 
+const getLogoUri = async (l1TokenAddress: string, zapperLogoUris: any) => {
+  const l1TokenAddressLCase = l1TokenAddress.toLowerCase()
+  const zapperUri = zapperLogoUris[l1TokenAddressLCase]
+
+  if (zapperUri) {
+    try {
+      const res = await axios.get(zapperUri)
+      if (res.status === 200) {
+        return zapperUri
+      }
+    } catch (e) {
+      // zapper uri not found
+    }
+  }
+  const trustWalletUri = `https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/${l1TokenAddress}/logo.png`
+
+  try {
+    const res = await axios.get(trustWalletUri)
+    if (res.status === 200) {
+      return trustWalletUri
+    }
+  } catch (e) {
+    // trustwallet uri not found
+  }
+  console.log('Could not get icon for', l1TokenAddress)
+
+  return
+}
+
+const getL1Name = (ERC20: ERC20) => {
+  switch (ERC20.address) {
+    case '0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2':
+      return 'Maker'
+    case '0x431ad2ff6a9C365805eBaD47Ee021148d6f7DBe0':
+      return 'dForce'
+    default:
+      return ERC20.name()
+  }
+}
```

Comment:
I think we should actually parse these as bytes32 strings instead of hardcoding specific addresses.
https://github.com/ethers-io/ethers.js/blob/ce8f1e4015c0f27bf178238770b1325136e3351a/packages/strings/src.ts/bytes32.ts#L21
```suggestion
const getL1Name = async (ERC20: ERC20) => {
   const name = await ERC20.name()
   if(name.length == 66) return ethers.utils.parseBytes32String(name)
   else return name
  }
}
```
same for the symbol

Reply:
also instead of `name.length == 66` we could `arrayify(name).length == 32` to express it in bytes